vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleRegistrationSpreadsheet.java

```diff
@@ -398,42 +405,78 @@ public List<SamplesheetHeaderName> retrieveGenomics() {
-      Cell commentCell = row.getCell(header.indexOf(SamplesheetHeaderName.CUSTOMER_COMMENT));
-
-      Supplier<Stream<Cell>> mandatoryCellStreamSupplier = () -> Stream.of(analysisTypeCell,
-          sampleLabelCell,
-          replicateIDCell, conditionCell, speciesCell, specimenCell, analyteCell);
+    Set<String> uniqueSamples = new HashSet<>();
+    for (int rowId = 1; rowId < sampleRegistrationSheet.getLastRowNum(); rowId++) {
+      Row row = sampleRegistrationSheet.getRow(rowId);
+
+      String analysisTypeInput = parseAndTrimCellOfRow(row, header.indexOf(SamplesheetHeaderName.SEQ_ANALYSIS_TYPE));
+      String sampleLabelInput = parseAndTrimCellOfRow(row, header.indexOf(SamplesheetHeaderName.SAMPLE_LABEL));
+      String replicateIDInput = parseAndTrimCellOfRow(row, header.indexOf(SamplesheetHeaderName.BIOLOGICAL_REPLICATE_ID));
+      String conditionInput = parseAndTrimCellOfRow(row, header.indexOf(SamplesheetHeaderName.CONDITION));
+      String speciesInput = parseAndTrimCellOfRow(row, header.indexOf(SamplesheetHeaderName.SPECIES));
+      String specimenInput = parseAndTrimCellOfRow(row, header.indexOf(SamplesheetHeaderName.SPECIMEN));
+      String analyteInput = parseAndTrimCellOfRow(row, header.indexOf(SamplesheetHeaderName.ANALYTE));
+      String commentInput = parseAndTrimCellOfRow(row, header.indexOf(SamplesheetHeaderName.CUSTOMER_COMMENT));
+
+      // we need to stream this list twice, so we use a supplier
+      Supplier<Stream<String>> mandatoryCellStreamSupplier = () -> Stream.of(analysisTypeInput,
+          sampleLabelInput,
+          replicateIDInput, conditionInput, speciesInput, specimenInput, analyteInput);
 
       if (mandatoryCellStreamSupplier.get().anyMatch(Objects::isNull)) {
         break;
       }
-      if (mandatoryCellStreamSupplier.get().noneMatch(x -> x.getStringCellValue().isEmpty())) {
-        ExperimentalGroup experimentalGroup = experimentalGroupToConditionString.get(
-            conditionCell.getStringCellValue().trim());
-        Long experimentalGroupId = experimentalGroup.id();
-        BiologicalReplicateId biologicalReplicateId = retrieveBiologicalReplicateId(
-            replicateIDCell.getStringCellValue().trim(), conditionCell.getStringCellValue().trim());
-        rows.add(new NGSRowDTO(analysisTypeCell.getStringCellValue().trim(),
-            sampleLabelCell.getStringCellValue().trim(),
-            biologicalReplicateId, experimentalGroupId,
-            speciesCell.getStringCellValue().trim(), specimenCell.getStringCellValue().trim(),
-            analyteCell.getStringCellValue().trim(),
-            commentCell.getStringCellValue().trim()));
+
+      if(mandatoryCellStreamSupplier.get().noneMatch(x -> x.isEmpty())) {
+        String uniqueSampleString = replicateIDInput+conditionInput;
+        if(uniqueSamples.contains(uniqueSampleString)) {
+          return Result.fromError(new SpreadsheetValidationException(
```

Comment:
As mentioned before, please think about how to avoid having exceptions in results. Think of a different way to return it (for example row id).